### DISCUSSION TITLE
Version 1.6 only works on the windows pod but not linux pod

### DISCUSSION THREAD

**Discussion by tphkit** [kubernetes-csi/csi-driver-smb#476] [General]
Hi there,

I managed to create the storageclass and pvc; also i could see a pv- folder is created in my smb server.  However, I could only see the files saved in the windows pod but not Linux pod.  Is  After some searching, the files are only saved in the node instead in the smb server for those Linux pod.  Any hints please?  Thanks!

**Comment by andyzhangx** [comment]
what's the "mount | grep cifs" output on the linux node? You may need to collect driver logs on the agent node: https://github.com/kubernetes-csi/csi-driver-smb/blob/master/docs/csi-debug.md#case2-volume-mountunmount-failed

**Reply by tphkit** [reply]
nothing from "mount | grep cifs"

Here is the driver log.  Many Thanks!

I0505 07:57:05.802341       1 utils.go:76] GRPC call: /csi.v1.Node/NodeStageVolume
I0505 07:57:05.802403       1 utils.go:77] GRPC request: {"secrets":"***stripped***","staging_target_path":"/var/snap/microk8s/common/var/lib/kubelet/plugins/kubernetes.io/csi/pv/pvc-bd50bf62-76a5-497b-9eb5-77b675af36ff/globalmount","volume_capability":{"AccessType":{"Mount":{"mount_flags":["dir_mode=0777","file_mode=0777","uid=1001","gid=1001","noperm","mfsymlinks","cache=strict","noserverino"]}},"access_mode":{"mode":5}},"volume_context":{"source":"//10.128.37.225/k8s01-pv","storage.kubernetes.io/csiProvisionerIdentity":"1651728576097-8081-smb.csi.k8s.io","subdir":"pvc-bd50bf62-76a5-497b-9eb5-77b675af36ff"},"volume_id":"10.128.37.225/k8s01-pv#pvc-bd50bf62-76a5-497b-9eb5-77b675af36ff#"}
I0505 07:57:05.803057       1 nodeserver.go:194] NodeStageVolume: targetPath(/var/snap/microk8s/common/var/lib/kubelet/plugins/kubernetes.io/csi/pv/pvc-bd50bf62-76a5-497b-9eb5-77b675af36ff/globalmount) volumeID(10.128.37.225/k8s01-pv#pvc-bd50bf62-76a5-497b-9eb5-77b675af36ff#) context(map[source://10.128.37.225/k8s01-pv storage.kubernetes.io/csiProvisionerIdentity:1651728576097-8081-smb.csi.k8s.io subdir:pvc-bd50bf62-76a5-497b-9eb5-77b675af36ff]) mountflags([dir_mode=0777 file_mode=0777 uid=1001 gid=1001 noperm mfsymlinks cache=strict noserverino]) mountOptions([dir_mode=0777 file_mode=0777 uid=1001 gid=1001 noperm mfsymlinks cache=strict noserverino])
I0505 07:57:05.803931       1 mount_linux.go:183] Mounting cmd (mount) with arguments (-t cifs -o dir_mode=0777,file_mode=0777,uid=1001,gid=1001,noperm,mfsymlinks,cache=strict,noserverino,<masked> //10.128.37.225/k8s01-pv/pvc-bd50bf62-76a5-497b-9eb5-77b675af36ff /var/snap/microk8s/common/var/lib/kubelet/plugins/kubernetes.io/csi/pv/pvc-bd50bf62-76a5-497b-9eb5-77b675af36ff/globalmount)
I0505 07:57:05.832592       1 nodeserver.go:223] volume(10.128.37.225/k8s01-pv#pvc-bd50bf62-76a5-497b-9eb5-77b675af36ff#) mount "//10.128.37.225/k8s01-pv/pvc-bd50bf62-76a5-497b-9eb5-77b675af36ff" on "/var/snap/microk8s/common/var/lib/kubelet/plugins/kubernetes.io/csi/pv/pvc-bd50bf62-76a5-497b-9eb5-77b675af36ff/globalmount" succeeded
I0505 07:57:05.832675       1 utils.go:83] GRPC response: {}
I0505 07:57:05.855805       1 utils.go:76] GRPC call: /csi.v1.Node/NodePublishVolume
I0505 07:57:05.855853       1 utils.go:77] GRPC request: {"staging_target_path":"/var/snap/microk8s/common/var/lib/kubelet/plugins/kubernetes.io/csi/pv/pvc-bd50bf62-76a5-497b-9eb5-77b675af36ff/globalmount","target_path":"/var/snap/microk8s/common/var/lib/kubelet/pods/30791a86-b8dd-4808-985c-9a3bcf760977/volumes/kubernetes.io~csi/pvc-bd50bf62-76a5-497b-9eb5-77b675af36ff/mount","volume_capability":{"AccessType":{"Mount":{"mount_flags":["dir_mode=0777","file_mode=0777","uid=1001","gid=1001","noperm","mfsymlinks","cache=strict","noserverino"]}},"access_mode":{"mode":5}},"volume_context":{"csi.storage.k8s.io/ephemeral":"false","csi.storage.k8s.io/pod.name":"nginx02-766757d4b5-lcvqs","csi.storage.k8s.io/pod.namespace":"default","csi.storage.k8s.io/pod.uid":"30791a86-b8dd-4808-985c-9a3bcf760977","csi.storage.k8s.io/serviceAccount.name":"default","source":"//10.128.37.225/k8s01-pv","storage.kubernetes.io/csiProvisionerIdentity":"1651728576097-8081-smb.csi.k8s.io","subdir":"pvc-bd50bf62-76a5-497b-9eb5-77b675af36ff"},"volume_id":"10.128.37.225/k8s01-pv#pvc-bd50bf62-76a5-497b-9eb5-77b675af36ff#"}
I0505 07:57:05.858038       1 nodeserver.go:78] NodePublishVolume: mounting /var/snap/microk8s/common/var/lib/kubelet/plugins/kubernetes.io/csi/pv/pvc-bd50bf62-76a5-497b-9eb5-77b675af36ff/globalmount at /var/snap/microk8s/common/var/lib/kubelet/pods/30791a86-b8dd-4808-985c-9a3bcf760977/volumes/kubernetes.io~csi/pvc-bd50bf62-76a5-497b-9eb5-77b675af36ff/mount with mountOptions: [bind] volumeID(10.128.37.225/k8s01-pv#pvc-bd50bf62-76a5-497b-9eb5-77b675af36ff#)
I0505 07:57:05.858087       1 mount_linux.go:183] Mounting cmd (mount) with arguments ( -o bind /var/snap/microk8s/common/var/lib/kubelet/plugins/kubernetes.io/csi/pv/pvc-bd50bf62-76a5-497b-9eb5-77b675af36ff/globalmount /var/snap/microk8s/common/var/lib/kubelet/pods/30791a86-b8dd-4808-985c-9a3bcf760977/volumes/kubernetes.io~csi/pvc-bd50bf62-76a5-497b-9eb5-77b675af36ff/mount)
I0505 07:57:05.862307       1 mount_linux.go:183] Mounting cmd (mount) with arguments ( -o bind,remount /var/snap/microk8s/common/var/lib/kubelet/plugins/kubernetes.io/csi/pv/pvc-bd50bf62-76a5-497b-9eb5-77b675af36ff/globalmount /var/snap/microk8s/common/var/lib/kubelet/pods/30791a86-b8dd-4808-985c-9a3bcf760977/volumes/kubernetes.io~csi/pvc-bd50bf62-76a5-497b-9eb5-77b675af36ff/mount)
I0505 07:57:05.866298       1 nodeserver.go:85] NodePublishVolume: mount /var/snap/microk8s/common/var/lib/kubelet/plugins/kubernetes.io/csi/pv/pvc-bd50bf62-76a5-497b-9eb5-77b675af36ff/globalmount at /var/snap/microk8s/common/var/lib/kubelet/pods/30791a86-b8dd-4808-985c-9a3bcf760977/volumes/kubernetes.io~csi/pvc-bd50bf62-76a5-497b-9eb5-77b675af36ff/mount volumeID(10.128.37.225/k8s01-pv#pvc-bd50bf62-76a5-497b-9eb5-77b675af36ff#) successfully
I0505 07:57:05.866345       1 utils.go:83] GRPC response: {}

**Reply by andyzhangx** [reply]
what's the `/var/log/messages` and `dmesg` output? I think cifs mount failed silently though the command returned success. 
```
I0505 07:57:05.832592 1 nodeserver.go:223] volume(10.128.37.225/k8s01-pv#pvc-bd50bf62-76a5-497b-9eb5-77b675af36ff#) mount "//10.128.37.225/k8s01-pv/pvc-bd50bf62-76a5-497b-9eb5-77b675af36ff" on "/var/snap/microk8s/common/var/lib/kubelet/plugins/kubernetes.io/csi/pv/pvc-bd50bf62-76a5-497b-9eb5-77b675af36ff/globalmount" succeeded
```

**Reply by andyzhangx** [reply]
I remembered, there is a similar case which is caused by incorrect kubelet path: https://github.com/kubernetes-csi/csi-driver-smb/issues/443#issuecomment-1097648755

**Reply by tphkit** [reply]
Yes, I think it is the same issue.  Could you please indicate me how to fix it?

**Reply by andyzhangx** [reply]
just follow: https://github.com/kubernetes-csi/csi-driver-smb/issues/443#issuecomment-1097648755

**Reply by tphkit** [reply]
got it.  Thanks!
